src/model/writer.js

```diff
@@ -1016,6 +1016,35 @@ export default class Writer {
 		}
 	}
 
+	/**
+	 * Disables default gravity behaviour that tries to get attributes from nodes surrounding the caret. When gravity is
+	 * marked as overridden then attributes from the node before the caret won't be taken into consideration while
+	 * updating selection attributes.
+	 *
+	 * For the following model fragment:
+	 *
+	 * 		<$text bold="true" linkHref="url">bar[]</$text><$text bold="true">biz</$text>
+	 *
+	 * Selection attribute keys before override will be equal `[ 'bold', 'linkHref' ]`
```

Comment:
* Default gravity: selection will have the `bold` and `linkHref` attributes.
* Overridden gravity: ...